### PR TITLE
Multi network wallets

### PR DESCRIPTION
**Pull request by LiorRabin**
## Proposed changes

multi-network wallets phase 2

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Refactor or a modification (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: server/src/routes/api/v2/wallets.js [hunk 1]
```diff
@@ -67,7 +67,8 @@ router.put('/token/:walletAddress', auth.required, async (req, res) => {
  */
 router.get('/', auth.required, async (req, res, next) => {
   const { phoneNumber, accountAddress } = req.user
-  const userWallet = await UserWallet.findOne({ phoneNumber, accountAddress }, { contacts: 0, firebaseToken: 0 })
+  const network = req.query.network || 'fuse'
+  const userWallet = await UserWallet.findOne({ phoneNumber, accountAddress, networks: network }, { contacts: 0, firebaseToken: 0 })
```
Comment (leonprou) [inline server/src/routes/api/v2/wallets.js]: if `networks` is array you're comparing an array and string (['fuse'] == 'fuse'). 
This works in mongoose?

Comment (LiorRabin) [reply server/src/routes/api/v2/wallets.js]: It works, I tried it both in the compass itself as well as run the code locally.
Surprised me too...

File: server/src/routes/api/v2/wallets.js [hunk 2]
```diff
@@ -222,4 +223,31 @@ router.post('/backup', auth.required, async (req, res, next) => {
...
+  const userWallet = await UserWallet.findOne({ phoneNumber, accountAddress })
+  if (!userWallet) {
+    const msg = `User ${phoneNumber}, ${accountAddress} doesn't have a wallet account on fuse yet, cannot create on ${network}`
+    return res.status(400).json({ error: msg })
+  }
+  if (userWallet.networks.includes[network]) {
+    const msg = `User ${phoneNumber}, ${accountAddress} already has wallet account: ${userWallet.walletAddress} on ${network}`
+    return res.status(400).json({ error: msg })
+  }
+  const job = await agenda.now('createForeignWallet', { userWallet })
```
Comment (bejavu) [inline server/src/routes/api/v2/wallets.js]: Please add `correlationId` to the agenda job (optional body param)
see other routes (mainly admin ones)

File: server/src/tasks/wallet.js [hunk 3]
```diff
@@ -100,7 +100,33 @@ const setWalletOwner = withAccount(async (account, { walletAddress, newOwner },
   return receipt
 })
 
+const createForeignWallet = withAccount(async (account, { userWallet, ens = '' }, job) => {
+  const { createContract, createMethod, send } = createNetwork('foreign', account)
+  const owner = userWallet.accountAddress
+  const walletFactory = createContract(WalletFactoryABI, userWallet.walletFactoryCurrentAddress)
+  const method = createMethod(walletFactory, 'createCounterfactualWallet', owner, Object.values(userWallet.walletModules), ens, userWallet.salt)
```
Comment (bejavu) [inline server/src/tasks/wallet.js]: this supposed to be original params not correct ones so we can be sure that the wallet will have the same address

**Pull request merged**